### PR TITLE
New Sandplanet Ruin: Cave Base

### PR DESCRIPTION
**Pull request by Erikafox**
## About The Pull Request
Adds a new ruin to the sand planet ruin pool - a small installation built into a cave holding some explosive revelations. And a Big Gun.
And satchel charges which are snowflaked c4. 
Also reformats blackboxes mildly so they can start empty.
dreamseeker_1podgNhvaZ

dreamseeker_JDRbW73Vxg

StrongDMM-2024-09-29 23 12 47
StrongDMM-2024-09-29 23 12 48

## Why It's Good For The Game
New map + simmy content

## Changelog

:cl:
add: Transceivers have been picking up an intermittent radio signal coming from a nearby sand world. People with too much spare time - please check it out.
add: Satchel Charges. Exceedingly rare and decently explosive.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/ammunition/ballistic/sniper.dm [hunk 1]
```diff
@@ -20,3 +20,24 @@
 	desc = "A .50 BMG penetrator bullet casing."
 	bullet_skin = "ap"
 	projectile_type = /obj/projectile/bullet/p50/penetrator
+
+//14.5mm Crunch Gun
+/obj/item/ammo_casing/p145
```
Comment (thgvr) [inline code/modules/projectiles/ammunition/ballistic/sniper.dm]: Yeah... this probably won't be happening

File: code/modules/projectiles/gun.dm [hunk 2]
```diff
@@ -854,8 +854,26 @@
 		actual_angle -= 360
 	if(total_recoil > 0)
 		recoil_camera(user, total_recoil + 1, (total_recoil * recoil_backtime_multiplier)+1, total_recoil, actual_angle)
+		var/consequences_of_our_actions = (HAS_TRAIT(user, TRAIT_EASYLIMBDISABLE)? 2 : 4)
```
Comment (thgvr) [inline code/modules/projectiles/gun.dm]: This mechanic is... questionable. I don't see it really being beneficial - if it happens at all, it should only happen on un-wielded very large calibre firearms. But definitely not added in a ruin PR.

File: code/modules/projectiles/guns/ballistic/rifle.dm [hunk 3]
```diff
@@ -193,3 +193,46 @@
 	mag_type = /obj/item/ammo_box/magazine/internal/boltaction/polymer
 	can_be_sawn_off = FALSE
 	manufacturer = MANUFACTURER_NONE
+
+
+/obj/item/gun/ballistic/rifle/crunch_gun
```
Comment (thgvr) [inline code/modules/projectiles/guns/ballistic/rifle.dm]: No snowflake gun reskins.

File: code/modules/projectiles/projectile/bullets/sniper.dm [hunk 4]
```diff
@@ -7,12 +7,13 @@
 	knockdown = 100
 	dismemberment = 50
 	armour_penetration = 60
+	var/obj_damage_mod = 1.1
```
Comment (thgvr) [inline code/modules/projectiles/projectile/bullets/sniper.dm]: What is this change for?

File: code/modules/projectiles/projectile/bullets/sniper.dm [hunk 5]
```diff
@@ -43,3 +44,37 @@
 	icon_state = "gaussstrong"
 	damage = 25
 	range = 16
+
+//crung gun
+/obj/projectile/bullet/p145
```
Comment (thgvr) [inline code/modules/projectiles/projectile/bullets/sniper.dm]: Again, no snowflake guns/calibres for ruins.

File: code/modules/ruins/sandplanet_ruin_code/cave_base.dm [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+/obj/machinery/porta_turret/cave_base
+	faction = list("turret", "Forward_Ops_Post")
+	stun_projectile = /obj/projectile/beam/laser/heavylaser
+	stun_projectile_sound = 'sound/weapons/lasercannonfire.ogg'
+	lethal_projectile = /obj/projectile/beam/laser/heavylaser
+	lethal_projectile_sound = 'sound/weapons/lasercannonfire.ogg'
+
+//gut wrenching content
+
+/datum/preset_holoimage/hapless_ipc
```
Comment (thgvr) [inline code/modules/ruins/sandplanet_ruin_code/cave_base.dm:53]: These are pretty solid.

File: code/modules/projectiles/projectile/bullets/sniper.dm [hunk 7]
```diff
@@ -43,3 +44,37 @@
...
+	dismemberment = 5
+	obj_damage_mod = 3
+
+/obj/projectile/bullet/p145/he
+	name = "14.5mm High Explosive shot"
+	damage = 30 // most of the damage should be in the explosion
+	dismemberment = 5
+
+/obj/projectile/bullet/p145/he/on_hit(atom/target, blocked = 0)
+	explosion(target, 0, 1, 1, 1, flame_range = 2)
```
Comment (thgvr) [inline code/modules/projectiles/projectile/bullets/sniper.dm]: holy

File: code/modules/assembly/voice.dm [hunk 8]
```diff
@@ -103,6 +103,10 @@
 	. = ..()
 	listening = FALSE
 
+/obj/item/assembly/voice/preset
+	mode = 1
+	recorded = "jorkin it"
```
Comment (thgvr) [inline code/modules/assembly/voice.dm]: hmm

File: code/game/objects/items/grenades/plastic.dm [hunk 9]
```diff
@@ -125,3 +125,16 @@
 	item_state = "plasticx4"
 	directional = TRUE
 	boom_sizes = list(0, 2, 5)
+
+
+// x-com ufo defense high ex charge 1993
+/obj/item/grenade/c4/satchel_charge
```
Comment (thgvr) [inline code/game/objects/items/grenades/plastic.dm:136]: This thing just feels a bit cheap to me. It pretty much only rewards knowing whats in the ruin beforehand, or having been there before.

**Review by thgvr — DISMISSED** [review]

File: code/game/objects/items/grenades/plastic.dm (hunk 9 above)
Comment (Erikafox) [reply code/game/objects/items/grenades/plastic.dm:136]: I'm not entirely sure how to rework it while still keeping the intent of it being a gotcha trap (that admittedly, won't trigger too often)

File: code/modules/projectiles/projectile/bullets/sniper.dm (hunk 4 above)
Comment (Erikafox) [reply code/modules/projectiles/projectile/bullets/sniper.dm]: snowflaked gun (changing hardcoded object damage to a modifier). Will look again at it in another pr.

**Review by Gristlebee — COMMENTED** [review]
- Please add a picture of the ruin spawned in game.
- The turrets guarding the base are set to be non-lethal. Having them on lethal would add some bite. Some extra traps like mines could be a fitting addition as well for defences.
- The airlock controller detects the door to the generator/laundry room, which makes it not cycle properly. 
image
- Items mapped on the same tile as a file cabinet get placed inside it on map gen. It isn't a big deal, but something to keep in mind if this wasn't intentional.
image
image
- The satchel charge is really strong, managing to fullgib me in an elite hardsuit. Getting rid of the devastation range should make it more manageable
- You've made a ruin with a strong story and premise, but my main concern when I look at this ruin is what there is not much to explore considering it's size. There is no compelling reason to check out the side caves, and the main content of the ruin, the base is on the small side. Some small rewards or greebles to reward looking around in the side caves would be a nice addition, along with some extra rooms in the base.

**Comment by Erikafox** [comment]
I added an additional area to explore (although it is somewhat intentionally sparse).
Explosive value has been changed
Turrets are set to lethal and some mines have been placed along the caves.

**Review by Gristlebee — COMMENTED** [review]
Feedback after testing this ruin on live:
---
Place the mines further away from fauna. They're able to set them off before the players arrive.
image

No tank mines please, these are insanely strong and can't be defused.
image

Swap out all or most of the fusion mines for fragmentation or rusted mines. Fusion mines are pretty nasty en masse.

The targeting on the turrets isn't set correctly, and don't lock onto players.

Put in another concrete wall here and finish the corner. It looks a bit odd otherwise.
image

The satchel bomb trap needs some kind of tell it activated, and a delay before it goes off.

**Comment by Erikafox** [comment]
re-drafted until i Make Prebuilt Assemblies Work

**Comment by Erikafox** [comment]
it works now.

**Review by Gristlebee — COMMENTED** [review]
These two turrets dont have the proper area set and don't function.
image

I noticed as well while players were running through the ruin the turrets didn't target captains or certain heads of staff roles, you'll want to double check the targeting settings.

File: code/modules/assembly/voice.dm (hunk 8 above)
Comment (Sun-Soaked) [reply code/modules/assembly/voice.dm]: incredible

**Comment by meemofcourse** [comment]
fix checks ?

**Comment by Gristlebee** [comment]
I love gambling

**Comment by Erikafox** [comment]
image
COME ONNNN

**Comment by Gristlebee** [comment]
Aw dangit

**Comment by Sun-Soaked** [comment]
Let's go gambling!

**Pull request merged**